### PR TITLE
allow static build of postgres provider

### PR DESCRIPTION
**Pull request by PeterPetrik**
follow up for https://github.com/qgis/QGIS/pull/31534/files for postgres providers 

required for https://github.com/lutraconsulting/input/issues/482

### CODE REVIEW

File: src/providers/postgres/qgspostgresprovidergui.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class QgsPostgresProviderGuiMetadata: public QgsProviderGuiMetadata
+{
+  public:
+    QgsPostgresProviderGuiMetadata();
+
+    QList<QgsSourceSelectProvider *> sourceSelectProviders() override;
+    QList<QgsDataItemGuiProvider *> dataItemGuiProviders() override;
+    QList<QgsProjectStorageGuiProvider *> projectStorageGuiProviders() override;
+    void QgsPostgresProviderGuiMetadata::registerGui( QMainWindow *mainWindow ) override;
```
Comment (wonder-sk) [inline src/providers/postgres/qgspostgresprovidergui.h]: `QgsPostgresProviderGuiMetadata::` is not needed?

File: src/providers/postgres/qgspostgresdataitemguiprovider.cpp [hunk 2]
```diff
@@ -156,6 +157,22 @@ bool QgsPostgresDataItemGuiProvider::handleDrop( QgsDataItem *item, QgsDataItemG
   return false;
 }
 
+QWidget *QgsPostgresDataItemGuiProvider::createParamWidget( QgsDataItem *root, QgsDataItemGuiContext )
+{
+  QgsPGRootItem *pgRootItem = qobject_cast<QgsPGRootItem *>( root );
+  if ( pgRootItem != nullptr )
+  {
+    // return new QgsWMSSourceSelect( nullptr, nullptr, QgsProviderRegistry::WidgetMode::Manager );
```
Comment (wonder-sk) [inline src/providers/postgres/qgspostgresdataitemguiprovider.cpp]: remove commented code?

File: src/providers/postgres/CMakeLists.txt [hunk 3]
```diff
@@ -80,41 +81,66 @@ IF (WITH_GUI)
...
+ENDIF (WITH_GUI)
+
+IF (FORCE_STATIC_PROVIDERS)
+  # for (external) mobile apps to be able to pick up provider for linking
+  INSTALL (TARGETS postgresprovider_a ARCHIVE DESTINATION ${QGIS_PLUGIN_DIR})
+  IF (WITH_GUI)
+    INSTALL (TARGETS postgresprovider_gui_a ARCHIVE DESTINATION ${QGIS_PLUGIN_DIR})
+  ENDIF (WITH_GUI)
+ELSE (FORCE_STATIC_PROVIDERS)
+  # dynimally loaded module
```
Comment (wonder-sk) [inline src/providers/postgres/CMakeLists.txt]: typo dynamically

**Review by wonder-sk — COMMENTED** [review]
Looking good.

I wouldn't mind to have this merged even for 3.10 but if you feel it's risky we can leave it to 3.12.

**Comment by nyalldawson** [comment]
> I wouldn't mind to have this merged even for 3.10

Looks low risk to me too

**Pull request merged**